frontend/packages/ceph-storage-plugin/src/components/modals/storage-pool-modal/storage-pool-modal.tsx

```diff
@@ -87,6 +91,22 @@ export const StoragePoolModal = withHandlePromise((props: StoragePoolModalProps)
   const [isSubmit, setIsSubmit] = React.useState(false);
   const [timer, setTimer] = React.useState<NodeJS.Timer>(null);
   const [isArbiterCluster, setArbiterCluster] = React.useState(false);
+  const [pools, poolsLoaded, poolsLoadedError] = useK8sGet<ListKind<StoragePoolKind>>(
+    CephBlockPoolModel,
+    null,
+    CEPH_STORAGE_NAMESPACE,
+  );
+  const failureDomain: string = React.useMemo(
+    () =>
+      pools && poolsLoaded && !poolsLoadedError
+        ? pools?.items.find((blockPoolConfig) =>
+            blockPoolConfig?.metadata.ownerReferences?.find(
+              (ownerReference) => ownerReference.name === OCS_INTERNAL_CR_NAME,
+            ),
+          ).spec?.failureDomain
+        : '',
+    [pools, poolsLoaded, poolsLoadedError],
```

Comment:
Its required during form submission so , lets move it to `submit`
No need to memoize as well.